DZ-8/src/main/java/org/example/StudentsGroup.java

```diff
@@ -0,0 +1,98 @@
+package org.example;
+import java.util.List;
+
+public class StudentsGroup {
+    // Дані про групу, private
+    private Student leader; // Староста
+    private List<Student> students; // Список студентів
+    private List<String> tasks; // Список завдань
+
+    public StudentsGroup(Student leader, List<Student> students, List<String> tasks) {
+        // Перевірка, чи є староста серед студентів
+        if (students.contains(leader)) {
+            this.leader = leader;
+            this.students = students;
+            this.tasks = tasks;
+        } else {
+            // Якщо немає, то викидаємо виняток
+            throw new IllegalArgumentException("Leader MUST be in the students list.");
+        }
+    }
+
+    public Student getLeader() {
+        return leader;
+    }
+
+    public void setLeader(Student leader) {
+        // Перевірка, чи є новий староста серед студентів
+        if (students.contains(leader)) {
+            this.leader = leader;
+        } else {
+            // Якщо немає, то викидаємо виняток
+            throw new IllegalArgumentException("Leader MUST be in the students list.");
+        }
+    }
+
+    public List<Student> getStudents() {
+        return students;
+    }
+
+    public void setStudents(List<Student> students) {
```

Comment:
You don't need to set students list but just add one by one add method `addStudent(Student student)` and implement adding to list `this.students` inside